src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -1031,6 +1028,1438 @@ static void ReportField(FieldInfo field)
+                    // (4,47): error CS8657: Static member 'S.PE.set' cannot be marked 'readonly'.
+                    //     static object PE {          get; readonly set; }
+                    Diagnostic(ErrorCode.ERR_StaticMemberCantBeReadOnly, "set").WithArguments("S.PE.set").WithLocation(4, 47),
+                    // (5,47): error CS8657: Static member 'S.PL.set' cannot be marked 'readonly'.
+                    //     static object PL {          get; readonly set { field = value; } }
+                    Diagnostic(ErrorCode.ERR_StaticMemberCantBeReadOnly, "set").WithArguments("S.PL.set").WithLocation(5, 47));
+            }
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void RefReturning_01(bool useStruct, bool useRefReadOnly)
+        {
+            string type = useStruct ? "struct" : "class";
+            string refModifier = useRefReadOnly ? "ref readonly" : "ref         ";
+            string source = $$"""
+                {{type}} S
+                {
+                    {{refModifier}} object P1 { get; }
+                    {{refModifier}} object P2 { get => ref field; }
+                    {{refModifier}} object P3 { get => ref field; set; }
+                    {{refModifier}} object P4 { get => ref field; init; }
+                    {{refModifier}} object P5 { get => ref field; set { } }
+                    {{refModifier}} object P6 { get => ref field; init { } }
+                    {{refModifier}} object P7 { get => throw null; }
+                    {{refModifier}} object P8 { get => throw null; set; }
+                    {{refModifier}} object P9 { get => throw null; init; }
+                    {{refModifier}} object PC { get => throw null; set { _ = field; } }
+                    {{refModifier}} object PD { get => throw null; init { _ = field; } }
+                    {{refModifier}} object PE { get; set; }
+                    {{refModifier}} object PF { get; init; }
+                    {{refModifier}} object PG { get; set { } }
+                    {{refModifier}} object PH { get; init { } }
+                    {{refModifier}} object PI { set { _ = field; } }
+                    {{refModifier}} object PJ { init { _ = field; } }
+                }
+                """;
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net80);
+            comp.VerifyEmitDiagnostics(
+                // (3,25): error CS8145: Auto-implemented properties cannot return by reference
+                //     ref          object P1 { get; }
+                Diagnostic(ErrorCode.ERR_AutoPropertyCannotBeRefReturning, "P1").WithLocation(3, 25),
+                // (4,25): error CS8145: Auto-implemented properties cannot return by reference
+                //     ref          object P2 { get => ref field; }
+                Diagnostic(ErrorCode.ERR_AutoPropertyCannotBeRefReturning, "P2").WithLocation(4, 25),
```

Comment:
we spec'd that the `field` keyword can't be used on a ref-returning property right?